### PR TITLE
[Form] forward valid numeric values to transform()

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #30114
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
@Rubinum @dbrumann @syastrebov I could need your feedback here as you all have been involved in #24036 and #26781.

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php [hunk 1]
```diff
@@ -58,7 +64,7 @@ public function transform($value)
             if (!is_numeric($value)) {
                 throw new TransformationFailedException('Expected a numeric.');
             }
-            $value = (string) ($value / $this->divisor);
+            $value /= $this->divisor;
```
Comment (dbrumann) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php]: I have tried forcing a rounding issue with and without casting to string and also by reverse transforming either value and I can't find any differences in the behavior. I can't remember why we put the string cast there, but I assume it was just for safety. This change looks good to me.

**Review by dbrumann — APPROVED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/NumberToLocalizedStringTransformer.php [hunk 2]
```diff
@@ -78,6 +78,11 @@ class NumberToLocalizedStringTransformer implements DataTransformerInterface
 
     private $scale;
 
+    /**
+     * @param int       $scale
```
Comment (OskarStark) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/NumberToLocalizedStringTransformer.php]: You can pass null here, too

File: src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformerTest.php [hunk 3]
```diff
@@ -17,6 +17,18 @@
 
 class MoneyToLocalizedStringTransformerTest extends TestCase
 {
+    private $previousLocale;
+
+    protected function setUp()
+    {
+        setlocale(LC_ALL, '0');
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformerTest.php]: $this->previousLocale = ...?

**Review by OskarStark — APPROVED** [review]

**Review by vudaltsov — APPROVED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php (hunk 1 above)
Comment (syastrebov) [reply src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php]: @xabbuh checked the referenced issue and this fix, looks good to me, yes, doesn't seem casting to string was added on specific purpose here, so i think it should be fine.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**